File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.py
class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        max_distance = 0

        for i in range(len(nums)):
            current_distance = abs(nums[i] - nums[i - 1])
            max_distance = max(max_distance, current_distance)
        return max_distance
                

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/maximum-difference-between-adjacent-elements-in-a-circular-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxAdjacentDistance = function(nums) {
    let result = -Infinity;
    for (let i = 0; i < nums.length; i++) {
        let next = (i + 1) % nums.length;
        result = Math.max(result, Math.abs(nums[i] - nums[next]));
    }
    return result;
};